frame/fetch.ts

```diff
@@ -69,53 +71,106 @@ export async function tryFetchJson(url: string): Promise<FetchJsonResult> {
-      errorMessage: "Expected JSON MIME type but received none",
+      status: FetchStatus.VALIDATION_ERROR,
+      errorMessage: `Expected ${mimeTypeDescription} MIME type but received none`,
     };
   }
-  if (!/^(application\/([^;]*\+)?|text\/)json(;.*)?$/i.test(contentType)) {
+  if (!mimeTypeRegExp.test(contentType)) {
     return {
-      status: FetchJsonStatus.VALIDATION_ERROR,
-      errorMessage: "Expected JSON MIME type but received:",
+      status: FetchStatus.VALIDATION_ERROR,
+      errorMessage: `Expected ${mimeTypeDescription} MIME type but received:`,
       errorData: [contentType],
     };
   }
   const xAllowFledge = response.headers.get("X-Allow-FLEDGE");
   if (xAllowFledge === null) {
     return {
-      status: FetchJsonStatus.VALIDATION_ERROR,
+      status: FetchStatus.VALIDATION_ERROR,
       errorMessage: "Expected header X-Allow-FLEDGE: true but received none",
     };
   }
   if (!/^true$/i.test(xAllowFledge)) {
     return {
-      status: FetchJsonStatus.VALIDATION_ERROR,
+      status: FetchStatus.VALIDATION_ERROR,
       errorMessage: "Expected header X-Allow-FLEDGE: true but received:",
       errorData: [xAllowFledge],
     };
   }
+  return { status: FetchStatus.OK, value: response };
+}
+
+/**
+ * Makes an HTTP request to a URL that's supposed to serve a JSON response body,
+ * and returns the parsed response.
+ */
+export async function tryFetchJson(url: string): Promise<FetchResult<unknown>> {
+  const result = await tryFetch(
+    url,
+    "application/json",
+    // https://mimesniff.spec.whatwg.org/#json-mime-type
+    // Chrome's behavior deviates from the spec here; it only allows +json if
+    // the top-level type is application.
+    /^\s*(application\/([^;]*\+)?|text\/)json\s*(;.*)?$/i,
+    "JSON"
+  );
+  if (result.status !== FetchStatus.OK) {
+    return result;
+  }
+  const response = result.value;
   let value: unknown;
   try {
     value = await response.json();
   } catch (error: unknown) {
     if (error instanceof TypeError) {
-      return { status: FetchJsonStatus.NETWORK_ERROR };
+      return { status: FetchStatus.NETWORK_ERROR };
     } /* istanbul ignore else */ else if (error instanceof SyntaxError) {
       return {
-        status: FetchJsonStatus.VALIDATION_ERROR,
+        status: FetchStatus.VALIDATION_ERROR,
         errorMessage: error.message,
       };
     } else {
       throw error;
     }
   }
-  return { status: FetchJsonStatus.OK, value };
+  return { status: FetchStatus.OK, value };
+}
+
+/**
+ * Makes an HTTP request to a URL that's supposed to serve a JavaScript response
+ * body, and returns the script as a string.
+ */
+export async function tryFetchJavaScript(
+  url: string
+): Promise<FetchResult<string>> {
+  const result = await tryFetch(
+    url,
+    "application/javascript",
+    // https://mimesniff.spec.whatwg.org/#javascript-mime-type
+    /^\s*((application|text)\/(x-)?(jav|ecm)ascript|text\/(javascript1\.[0-5]|(j|live)script))\s*(;.*)?$/i,
```

Comment:
I guess this is OK, but I think if we just required `(application|text)/javascript` that would be fine too

Reply:
Seems better not to be more restrictive than Chrome/the actual Web standard if we don't have to be.